app/helpers/markdown_helper.rb

```diff
@@ -1,16 +1,44 @@
 module MarkdownHelper
   include CamoHelper
 
-  def camofy(markdown)
-    return unless markdown
+  def camofy(text)
+    return unless text
 
-    markdown.gsub(markdown_img_regex) do
+    text = sub_markdown(text)
+    sub_html(text)
+  end
+
+  def sub_markdown(text)
+    text.gsub(markdown_img_regex) do
       "![#{Regexp.last_match(1)}](#{camo(Regexp.last_match(2))}#{Regexp.last_match(3)})"
     end
   end
 
+  def sub_html(text)
+    text.gsub(html_img_regex) do
+      preceding_src = Regexp.last_match(1)
+      quote_mark = Regexp.last_match(2)
+      url = Regexp.last_match(3)
+      ending = Regexp.last_match(4)
+      "<img#{preceding_src} src=#{quote_mark}#{camo(url)}#{quote_mark}#{ending}"
+    end
+  end
+
   def markdown_img_regex
     # ![alt text](url =widthxheight "title")
     /!\[([^\[]*)\]\(([^\ )]+)(( =[^)]?[^ ]+)?( [^)]?"[^)]+")?)?\)/
   end
+
+  def html_img_regex
+    # warning: this regex may not be perfect. They rarely are.
+    # If you find an edge case, improve this regex!
+    # <img...something... src="url"...ending...
+    # or, the alternative quotes: <img...something... src='url'...ending...
+    # or, even without quotes: <img...something... src=url...ending...
+    # and the ...ending... can be either a space, a > or />
+    # note that we don't allow mismatched quotes like 'url" or shenanigans like that
+    # This regex contains two particularly useful features:
+    #  capturing groups, and lazy matching.
+    %r{<img([^>]*) src=(["']?)(.+?)\2( |>|/>)}
```

Comment:
```suggestion
   %r{<img[^>]*\s+src=(["']?)([^'">]+)\1(?=[/>])}
```
this one also checks for space in between the elements and has better checking for the url

Reply:
you seem to have removed the first capturing group. chatgpt has that tendency because it doesn't see you using its result anywhere in your regex. However, we definitely use this capturing group to produce the new text. What I do like, is the use of `\s`; that's any whitespace right?

What exactly does the latter part improve? Because I don't think it really improves anything. Let me break my thoughts down:
- **original**: `(.+?)\2( |>|/>)` where `\2` matches to the same value as the captured opening quote (either ', " or nothing at all). `(.+?)` _lazily_ matches any character. That means that it will begin with `src='` or `src="` or `src=` and then it will continue to match any character until the first time it encounters that same quotation mark again. Subsequently, after the quotation mark, it needs to match one of the three options in `( |>|/>)` which are a space, the `>` or `/>`.
- **your suggestion**: `([^'">]+)\1(?=[/>])}` where `\1` matches the same value as the captured opening quote. Note that `[^'">]` is unnecessarily restrictive: I don't know exactly the specification of HTML, but I can image that something like `src='lookatthisdoublequote"isntitamazing'` is valid. Notice the `"` in the middle? Your regex suggestion would stop the src value at that middle quote, because it does not pass the `[^'"]` check. And the latter part, `(?=[/>])` does not allow for any whitespace.


I do have some ideas based on your suggestion:
- use `\s` instead of spaces when I'm indicating whitespace
- I've reconsidered whether the last capturing group is necessary, but yes it is, because when we have `src=somesource`, we can only determine the end of the source value when we encounter a whitespace or / or >. But I can change it from `( |>|/>)` to `( |>|/)` which can be put simpler by writing `[ >/]`

Reply:
I've applied the new ideas I mentioned in the latest commit